src/components/formPost/formPost.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from 'react';
+
+type User = {
+  id: number;
+  name: string;
+  username: string;
+  email: string;
+};
+
+type ToDo = {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number | null;
+  user: User | null;
+};
+
+type Props = {
+  onSubmit: (newAdd: ToDo) => void;
+  users: User[];
+  heighestId: number;
+};
+
+export const FormPost: React.FC<Props> = ({ onSubmit, users, heighestId }) => {
+  const [value, setValue] = useState<string>('');
+  const [touch, setTouch] = useState<boolean>(false);
+  const [touchSelect, setTouchSelect] = useState<boolean>(false);
+  const [select, setSelect] = useState<string>('');
+  const reset = (): void => {
+    setValue('');
+    setSelect('');
+  };
+
+  const handleChangeSelect = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelect(e.currentTarget.value);
+    setTouchSelect(false);
+  };
+
+  const handleChangeInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setValue(e.currentTarget.value);
+    if (e.currentTarget.value) {
+      setTouch(false);
+    }
+  };
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    setTouch(!value);
+    setTouchSelect(!select);
+
+    if (select && value) {
+      const foundUser: User | null =
+        users.find(user => user.name === select) || null;
+      const userId: User | null =
+        users.find(user => user.name === select) || null;
```

Comment:
The variable `userId` is redundant as it repeats the logic of `foundUser`. You can directly use `foundUser?.id` in the `onSubmit` call.